pandas/tests/frame/methods/test_astype.py

```diff
@@ -136,6 +137,15 @@ def test_astype_cast_nan_inf_int(self, val, dtype):
         with pytest.raises(ValueError, match=msg):
             df.astype(dtype)
 
+    @pytest.mark.parametrize("dtype", ["Int64", "Int32", "Int16"])
+    @pytest.mark.parametrize("val", [np.nan, NA])
+    def test_astype_cast_via_loc_nan_int(self, val, dtype):
+        # GH31861
```

Comment:
can you add the working conditions as well (e.g. nan + a value) from the OP

Reply:
sure! let me try.